### ISSUE TITLE
Cannot work with multiple files with same reference blocks

### ISSUE THREAD

**Issue by gera2ld** [zont/gulp-usemin#110]
Similar to #91 , I am wondering whether there is a way to work with multiple files with the same reference blocks. For example:

1.html:
``` html
<html>
<head>
<!-- build:css style.css -->
<link rel="stylesheet" href="style.less" />
<!-- endbuild -->
</head>
<body>
This is 1.html
</body>
</html>
```

2.html:
``` html
<html>
<head>
<!-- build:css style.css -->
<link rel="stylesheet" href="style.less" />
<!-- endbuild -->
</head>
<body>
This is 2.html
</body>
</html>
```

v0.3.11 does not work for me. As a result, *style.less* is piped twice and an error is thrown out in the second time because the pipeline is ended after the first time.

I think the groups of references can be identified by the destination names. When the same name appears in another file, just replace the references and ignore the files referred.

BTW, I can see many developers expecting pipelines to be reused, which I think is a very good idea. In that way we can define a pipeline that several groups of files will go through (the [lazypipe](https://www.npmjs.com/package/lazypipe) way), instead of writing pipelines with the same procedures again and again.

**Comment by ryanlelek** [comment]
+1 - Here because I expected the configuration to be reused on multiple files

**Comment by gera2ld** [comment]
I think [gulp-inject](https://www.npmjs.com/package/gulp-inject) satisfies me in such cases.

**Comment by ryanlelek** [comment]
Fellow Google travelers...
Check out gera2ld's answer

What worked for me:

Gulp ForEach:
https://github.com/mariusGundersen/gulp-forEach
```
gulp.task('your-regular-task', function () {
  return gulp.src('src/*.html')
    // 1) Move your Code that was HERE
    // New Module
    .pipe(foreach(function (stream, file) {
      return stream
        // 2) Move to here
        .pipe(usemin({
          html    : [minify_html({ quotes : true, empty : true })],
          css_app : [minify_css({ keepSpecialComments : 1 }), 'concat'],
          js_app  : [minify_js(), 'concat']
        }))
        // BE CAREFUL!!!
        // This now has the CSS/JS files added to the stream
        // Not just the HTML files you sourced
        .pipe(gulp.dest('build/'));
    }))
});
```

**Comment by gera2ld** [comment]
This is a good idea to reuse the pipes but not the exact way I expect. Because when I mentioned *same reference blocks* I mean they share the same files which need to be piped to *usemin* only once. But, as far as I know, gulp-foreach will pipe each block to *usemin* once, and the files in those blocks will be *usemin*'d repeatedly.


 在2015年5月13日 10:45:08, "Ryan Lelek" <notifications@github.com> 写道：



Fellow Google travelers...
Check out gera2ld's answer

What worked for me:

Gulp ForEach:
https://github.com/mariusGundersen/gulp-forEach

gulp.task('your-regular-task', function () {
  return gulp.src('src/*.html')
    // 1) Move your Code that was HERE
    // New Module
    .pipe(foreach(function (stream, file) {
      return stream
        // 2) Move to here
        .pipe(usemin({
          html    : [minify_html({ quotes : true, empty : true })],
          css_app : [minify_css({ keepSpecialComments : 1 }), 'concat'],
          js_app  : [minify_js(), 'concat']
        }))
        // BE CAREFUL!!!
        // This now has the CSS/JS files added to the stream
        // Not just the HTML files you sourced
        .pipe(gulp.dest('build/'));
    }))
});

**Comment by zont** [comment]
Fixed in 0.3.14

**Comment by hoyangtsai** [comment]
still happening in 0.3.14
